Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -579,16 +554,51 @@ public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetBy
             .ToPagedResultAsync(DefaultSubmissionsPerPage, page);
     }
 
-    public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetPendingSubmissions(int page)
-        => await this.submissionsCommonData
-            .GetAllPending()
-            .OrderByDescending(s => s.Id)
-            .MapCollection<SubmissionForPublicSubmissionsServiceModel>()
-            .ToPagedResultAsync(DefaultSubmissionsPerPage, page);
-
     public Task<int> GetTotalCount()
         => this.submissionsData.GetTotalSubmissionsCount();
 
+    public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetSubmissions(SubmissionEnumType type, int page)
+    {
+        if (type == SubmissionEnumType.Public)
+        {
+            var user = this.userProviderService.GetCurrentUser();
+
+            if (user.IsAdminOrLecturer)
+            {
+                return await this.submissionsData.GetLatestSubmissions<SubmissionForPublicSubmissionsServiceModel>(
+                    DefaultSubmissionsPerPage, page);
+            }
+
+            var modelResult = new PagedResult<SubmissionForPublicSubmissionsServiceModel>();
+
+            modelResult.Items =
+                await this.submissionsData.GetLatestSubmissions<SubmissionForPublicSubmissionsServiceModel>(
+                    DefaultSubmissionsPerPage);
+
+            return modelResult;
+        }
+
+        if (type == SubmissionEnumType.Processing)
+        {
+            return await this.submissionsCommonData
+                .GetAllProcessing()
```

Comment:
You can make `GetAllProcessing<T>` generic and do the mapping in the methods. This will reduce the repetition of code here. Also this can be exported in a variable and you can invoke .ToPagedResultAsync(DefaultSubmissionsPerPage, page); in the return statement.

Reply:
Can you look at the new version of the code I’ve uploaded? Is it necessary to add GetAllProcessing<T>() now?